### PR TITLE
Installing GitHub App on an organization will automatically create a course

### PR DESCRIPTION
**Pull request by JosteinLindhom**
Installing a GitHub App on an organization will trigger QuickFeed to automatically create a course for the organization.

### CODE REVIEW

File: web/hooks/installation.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"context"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/quickfeed/quickfeed/qf"
+	"github.com/quickfeed/quickfeed/scm"
+)
+
+func (wh GitHubWebHook) handleInstallationCreated(event *github.InstallationEvent) {
+	installerID := uint64(event.GetSender().GetID())
+	user, err := wh.db.GetUserByRemoteIdentity(installerID)
```
Comment (meling) [inline web/hooks/installation.go:13]: `s/user/courseCreator/`.

Should this user be checked if it is admin and allowed to create courses, ref. `interceptor/access_control.go`:
```json
	"CreateCourse":           {admin},
```
There is also this in `web/interceptor/tokens.go`:
```go
	"CreateCourse": // The signed in user gets the teacher role in the new course.
	func(ctx context.Context, tm *auth.TokenManager, _ userIDs) error {
		claims, ok := auth.ClaimsFromContext(ctx)
		if !ok {
			return fmt.Errorf("CreateCourse: missing claims in context")
		}
		return tm.Add(claims.UserID)
	},
```
Or can we trust that the event is from GitHub and representing the course creator?

**Review by meling — CHANGES_REQUESTED** [review]
This is just a partial review; too tired to continue.

File: web/hooks/installation.go (hunk 1 above)
Comment (JosteinLindhom) [reply web/hooks/installation.go:13]: I believe we can trust that the event is coming from GitHub if we use [webhook secrets](https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries).

File: web/hooks/installation.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	if err := wh.db.CreateCourse(courseCreator.ID, course); err != nil {
+		wh.logger.Errorf("Could not create database record for course %s: %v", orgName, err)
+	}
+
+	if err := wh.tm.Add(courseCreator.ID); err != nil {
+		wh.logger.Errorf("Could not add user %s for token refresh: %v", courseCreator.Login, err)
+	}
+}
+
+func defaultYear() uint32 {
```
Comment (meling) [inline web/hooks/installation.go:76]: Add test.

File: web/hooks/installation.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	now := time.Now()
+	if now.Month() <= 11 && now.Day() <= 31 && now.Month() > 10 {
+		return uint32(now.Year() + 1)
+	}
+	return uint32(now.Year())
+}
+
+func defaultTag() string {
```
Comment (meling) [inline web/hooks/installation.go:84]: Add test

**Review by meling — CHANGES_REQUESTED** [review]

**Review by meling — APPROVED** [review]

**Pull request merged**